src/main/java/com/coveo/feign/ReflectionErrorDecoder.java

```diff
@@ -132,8 +132,15 @@ public Exception decode(String methodKey, Response response) {
 
   private void initialize() {
     try {
-      detailMessageField = Throwable.class.getDeclaredField("detailMessage");
-      detailMessageField.setAccessible(true);
+      if (Runtime.version().feature() < 15) {
```

Comment:
not sure why this wasn't done before but could you move this into a `static` block so it only runs once on class init? It's not running on each instance creation which doesn't make much sense.